pac4j-oauth/src/main/java/org/pac4j/oauth/profile/generic/GenericOAuth20ProfileDefinition.java

```diff
@@ -0,0 +1,117 @@
+package org.pac4j.oauth.profile.generic;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.Verb;
+import java.util.HashMap;
+import java.util.Map;
+import org.pac4j.core.exception.HttpAction;
+import org.pac4j.core.profile.converter.AttributeConverter;
+import org.pac4j.core.profile.converter.StringConverter;
+import org.pac4j.oauth.config.OAuth20Configuration;
+import org.pac4j.oauth.profile.JsonHelper;
+import org.pac4j.oauth.profile.definition.OAuth20ProfileDefinition;
+
+/**
+ * <p>This class is the user profile for generic OAuth2 with appropriate getters.</p>
+ * <p>The map of <code>profileAttributes</code> is intended to replate the primary/secondary attributes where
+ * the key is the name of the attribute and the value is the path to obtain that attribute from the
+ * json resopnse, starting from <code>firstNodePath</code></p>
+ * <p>It is returned by the {@link org.pac4j.oauth.client.GenericOAuth2Client}.</p>
+ *
+ * @author Julio Arrebola
+ */
+public class GenericOAuth20ProfileDefinition extends OAuth20ProfileDefinition<GenericOAuth20Profile> {
+
+    
+    private final Map<String,String> profileAttributes = new HashMap<>();
+    
+    String profileUrl = null;
+    String profileMethod = null;
+    String firstNodePath = null;
+    
+    public void setProfileMethod(String value) {
+        this.profileMethod = value;
+    }
+
+    @Override
+    public Verb getProfileVerb() {
+        if ("POST".equalsIgnoreCase(profileMethod)) {
+            return Verb.POST;
+        } else if ("GET".equalsIgnoreCase(profileMethod)) {
+            return Verb.GET;
+        } else {
+            return super.getProfileVerb(); 
+        }
+    }
+    
+    public void setProfileUrl(String profileUrl) {
+        this.profileUrl = profileUrl;
+    }
+
+    @Override
+    public String getProfileUrl(OAuth2AccessToken accessToken, OAuth20Configuration configuration) {
+        return profileUrl;
+    }
+
+    @Override
+    public GenericOAuth20Profile extractUserProfile(String body) throws HttpAction {
+        final GenericOAuth20Profile profile = new GenericOAuth20Profile();
+        final JsonNode json = JsonHelper.getFirstNode(body, getFirstNodePath());
+        if (json != null) {
+            for (final String attribute : getPrimaryAttributes()) {
+                convertAndAdd(profile, attribute, JsonHelper.getElement(json, attribute));
+            }
+            for (final String attribute : getSecondaryAttributes()) {
+                convertAndAdd(profile, attribute, JsonHelper.getElement(json, attribute));
+            }
+            for (final Map.Entry<String, String> entry : getProfileAttributes().entrySet()) {
```

Comment:
I found two issues with primary/secondary attributes:
-In this new implementation both kind of attributes collect its values in the same attributes map, so I wasn't sure both lists would be used in the future.
-Most important, since both are lists, the name of the attribute and the jsosn attribute containing its values have to be the same.

That's why I added a map of profile attributes where its key is the name of the attribute in the profile and its value is the reference to the json element containing its value.